doc/vxlan/EVPN/EVPN_VxLAN_Multihoming.md

```diff
@@ -0,0 +1,1450 @@
+     * @brief Number of L2 ECMP GROUP members in the group
+     *
+     * @type sai_uint32_t
+     * @flags READ_ONLY
+     */
+    SAI_L2_ECMP_GROUP_ATTR_MEMBER_COUNT = SAI_L2_ECMP_GROUP_ATTR_START,
+
+    /**
+     * @brief L2 ECMP GROUP member list
+     *
+     * @type sai_object_list_t
+     * @flags READ_ONLY
+     * @objects SAI_OBJECT_TYPE_L2_ECMP_GROUP_MEMBER
+     */
+    SAI_L2_ECMP_GROUP_ATTR_MEMBER_LIST,
+
+    /**
+     * @brief Attach a counter
+     *
+     * When it is empty, then packet hits won't be counted
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_COUNTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_L2_ECMP_GROUP_ATTR_COUNTER_ID,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_L2_ECMP_GROUP_ATTR_END,
+
+    /** Custom range base value */
+    SAI_L2_ECMP_GROUP_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_L2_ECMP_GROUP_ATTR_CUSTOM_RANGE_END
+
+} sai_l2_ecmp_group_attr_t;
+
+```
+
+- **L2_ECMP_GROUP_MEMBER**
+  - Represents a remote VTEP.
+  - Is associated with an L2 ECMP GROUP.
+
+```
+typedef enum _sai_l2_ecmp_group_member_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_L2_ECMP_GROUP_MEMBER_ATTR_START,
+
+    /**
+     * @brief L2 ECMP GROUP id
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_L2_ECMP_GROUP
+     */
+    SAI_L2_ECMP_GROUP_MEMBER_ATTR_L2_ECMP_GROUP_ID = SAI_L2_ECMP_GROUP_MEMBER_ATTR_START,
+
+    /**
+     * @brief P2P Tunnel oid
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_TUNNEL
+     */
+    SAI_L2_ECMP_GROUP_MEMBER_ATTR_TUNNEL_ID,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_L2_ECMP_GROUP_MEMBER_ATTR_END,
+
+    /** Custom range base value */
+    SAI_L2_ECMP_GROUP_MEMBER_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_L2_ECMP_GROUP_MEMBER_ATTR_CUSTOM_RANGE_END
+
+} sai_l2_ecmp_group_member_attr_t;
+
+```
+
+- Capabilities corresponding to the above new SAI objects to be implemented.
+
+
+### 3.4.2 Changes to Existing SAI Objects
+
+- **BRIDGEPORT**
+  - A new bridge-port type of type L2_ECMP_GROUP
+  - A new attribute NON_DF to drop BUM traffic .
+    - Applicable to bridgeport type PORT.
+    - When set, egress BUM traffic on this bridge port will be dropped.
+    - Traffic dropped due to this should not be counted against SAI_PORT_STAT_IF_OUT_DISCARDS i.e Interface Tx drop counters.
+```
+    /**
+     * @brief Indication if this bridge port is Not a Designated Forwarder
+     * When set, egress BUM traffic on this bridge port will be dropped.
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     * @validonly SAI_BRIDGE_PORT_ATTR_TYPE == SAI_BRIDGE_PORT_TYPE_PORT
+     */
+    SAI_BRIDGE_PORT_ATTR_NON_DF,
```

Comment:
It is for access facing ports (ESI configured interfaces.)

Reply:
How does SAI knows whether a PORT is part of Ethernet segment or not..? do we one more attribute to define that. I think, SAI can't depend on (SAI_BRIDGE_PORT_ATTR_NON_DF) attribute, since this is FALSE even for both Ethernet segment and single homed switches.

and second doubt is, does this default==false condition causes any Micro-loops in the network..?
Consider a case for PO2 configuration..
1. when an acess PORT with Ethernet segment enabled is added to Valn, SAI calls bridge port with (SAI_BRIDGE_PORT_ATTR_NON_DF== false), since still DF election is in progress.
2. above condition is smae on all 4 VTEPs. Now all the nodes which are part of PO2(VTEP-1,2,3,4) will start flooding all the BUM traffic.
3. after the DF election is completed, SAI will update the NON-DF nodes with (SAI_BRIDGE_PORT_ATTR_NON_DF= true)
4. till this time, traffic is flooding on all VTEPs. does this momentary flooded traffic will not cause transient loops in the network..?